### PR TITLE
Backport v1.3 - Pull request 6970

### PR DESCRIPTION
**Pull request by eloycoto**
1) Two commits are almost original
2) One commit is a complete new commit. 

#6970

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7077)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by ianvernon** [comment]
@eloycoto unit tests failed due to compilation errors: 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/9871/

```
10:30:12 ok  	github.com/cilium/cilium/common/addressing	0.009s	coverage: 87.6% of statements
10:30:12 ?   	github.com/cilium/cilium/common/types	[no test files]
10:30:31 # github.com/cilium/cilium/daemon
10:30:31 daemon/loadbalancer.go:641: undefined: loadbalancer.AllProtocols
10:30:31 FAIL	github.com/cilium/cilium/daemon [build failed]
10:30:31 make: *** [tests-ginkgo-real] Error 1
10:30:31 Makefile:50: recipe for target 'tests-ginkgo-real' failed
10:30:31 Makefile:36: recipe for target 'tests-ginkgo' failed
10:30:31 make: *** [tests-ginkgo] Error 2
```

**Comment by eloycoto** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-missed-k8s

File: pkg/maps/lbmap/lbmap.go [hunk 1]
```diff
@@ -548,12 +548,12 @@ func serviceKeynValue2FEnBE(svcKey ServiceKey, svcValue ServiceValue) (*loadbala
 
 // RevNat6Value2L3n4Addr converts the given RevNat6Value to a L3n4Addr.
```
Comment (ianvernon) [inline pkg/maps/lbmap/lbmap.go:549]: @eloycoto there was a change in 
https://github.com/cilium/cilium/commit/77d07735cf0b9a05a7156fc693a3a94866956709 to function `serviceKeynValue2FEnBE` which is not present here - is that intentional?

**Review by ianvernon — COMMENTED** [review]
Looks good since the tests passed, but do have one question about something that was left out from one of the backported commits.

File: daemon/loadbalancer.go [hunk 2]
```diff
@@ -628,6 +628,31 @@ func (d *Daemon) SyncLBMap() error {
 	return nil
 }
 
+// FrontendList is the list of all k8s service frontends
```
Comment (tgraf) [inline daemon/loadbalancer.go:631]: Use `k8s: Add LooseMatch() function to validate service presence` as the commit message as `Backport commit` does not mean anything

File: pkg/maps/lbmap/lbmap.go (hunk 1 above)
Comment (tgraf) [reply pkg/maps/lbmap/lbmap.go:549]: Seems missing. It's only cosmetic right now but a future backport might require that frontend and backend have matching protocol values.

**Review by tgraf — CHANGES_REQUESTED** [review]
Custom backport itself look. Agreed with @ianvernon on missing chunk

**Comment by eloycoto** [comment]
@ianvernon good catch, changes made

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-missed-k8s

**Comment by eloycoto** [comment]
test-missed-k8s

**Comment by eloycoto** [comment]
test-docs-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**